Stats: Add image to upgrade notice

a8ck3n opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Part of #28729.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->

Adds an image + gradient to the upgrade nudge.

<img width="860" alt="SCR-20230306-keh" src="https://user-images.githubusercontent.com/40267301/223037270-9f86d5aa-06af-4694-8a30-4c138d10e3d7.png">

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

- Make sure Odyssey Stats are disabled.
- Visit Jetpack → Stats.
- Confirm the image is present in the upgrade nudge.
- Confirm the image is removed at smaller screen sizes.

**Looking for feedback!**

I'm not super happy with the gap below the image when resizing the screen but I'm unsure of how to remove it and have the image size itself properly. I've tried setting both images via CSS but then neither are drawn. I need something in the HTML to trigger proper layout.

**Updated:(per feedback)**

- Images are now handled strictly via CSS.
- Font sizes and padding adjusted.
- Missing translation calls have been added back.
- Updated image in PR description to match current status of the notice.

a8ck3n committed:
[not verified] Add images for use in upgrade notice.

a8ck3n committed:
[not verified] Updated HTML & CSS.

a8ck3n committed:
[not verified] Remove old notice HTML & CSS.

a8ck3n committed:
[not verified] Restore conditional display of notice.

a8ck3n committed:
Remove unused CSS bits.

a8ck3n committed:
Update class names and CSS selectors.

a8ck3n committed:
Hide image on smaller screens.

a8ck3n committed:
Changelog++

a8ck3n committed:
Downsize image & gradient.

a8ck3n committed:
Update max-height of image.

jsnmoon commented on projects/plugins/jetpack/modules/stats.php:
```diff
@@ -833,18 +853,21 @@ function stats_print_odyssey_nudge( $html ) {
-				<button class="dops-button stats-odyssey-nudge--primary-button">
+	<div id="stats-odyssey-nudge-main" class="stats-odyssey-notice">
+		<div class="stats-odyssey-notice--content">
+			<h2 class="stats-odyssey-notice--content-header">Explore the new Jetpack Stats</h2>
+			<p class="stats-odyssey-notice--content-text">We've added new stats and insights in a more modern and mobile friendly experience to help you grow your site.</p>
```
What's the reason for removing the `esc_html_e` translation calls for these two lines?

a8ck3n replied:
Ah! No reasoning behind this. Just a mistake on my part. I was writing the new layouts side-by-side with the old ones (ie: a new function call) and displaying both in the UI so I could compare the results. I overlooked this when cleaning up and removing the old code path.

Thanks for flagging this!

jsnmoon commented on projects/plugins/jetpack/modules/stats.php:
```diff
@@ -763,48 +763,67 @@ function stats_print_odyssey_nudge( $html ) {
 
 			border: 1px solid var( --jp-gray-5 );
 			border-radius: 4px;
 			border-color: var( --jp-black );
```
```suggestion
			border-color: var( --jp-gray-5 );
			border-left-color: var( --jp-black );
```

jsnmoon commented on projects/plugins/jetpack/modules/stats.php:
```diff
@@ -763,48 +763,67 @@ function stats_print_odyssey_nudge( $html ) {
-		.stats-odyssey-nudge--content {
-			flex-grow: 1;
+		.stats-odyssey-notice--content {
+			padding: 24px 0 27px 18px;
```
<img width="660" alt="image" src="https://user-images.githubusercontent.com/4044428/222835118-f8450137-7205-46f0-bbb0-105c967a4fd3.png">

```suggestion
			padding: 24px 0 24px 30px;
```

jsnmoon commented on projects/plugins/jetpack/modules/stats.php:
```diff
@@ -763,48 +763,67 @@ function stats_print_odyssey_nudge( $html ) {
-		.stats-odyssey-nudge--title {
-			font-weight: 600;
+		.stats-odyssey-notice--content-header {
+			font-size: 16px;
```
<img width="1231" alt="image" src="https://user-images.githubusercontent.com/4044428/222836268-ee46a9d3-33e8-497c-83b4-f8dbee112ce0.png">

```suggestion
			font-size: 24px;
			line-height: 32px;
```

jsnmoon reviewed (changes requested):
These changes look reasonable and test well on my local Jetpack test site. I recommended some styling tweaks and highlighted where we might have accidentally removed two internationalization calls.

I do have a suggestion for simplifying [our background implementation](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Backgrounds_and_Borders/Using_multiple_backgrounds), however:

1. Remove `img.stats-odyssey-notice--image`.
2. Apply `width: 100%;` to both `.stats-odyssey-notice--content` and `.stats-odyssey-notice--image-container`.
3. Replace the existing `background-image` rule with `url("/wp-content/plugins/jetpack/images/odyssey-upgrade/background.png"), url("/wp-content/plugins/jetpack/images/odyssey-upgrade/gradient.png")`.

The result should look like the following:

<img width="1019" alt="image" src="https://user-images.githubusercontent.com/4044428/222837473-f2af43a8-b792-4204-bb40-ce8b7c9f256b.png">

a8ck3n committed:
Update notice to use CSS background images.

a8ck3n committed:
Add back missing translation calls.

a8ck3n committed:
Some CSS tweaks to better match initial design.

a8ck3n commented:
Thanks for the review and suggestions @jsnmoon. Changes incoming!

Regarding the padding and font bits, I was working from the [Notice class](https://github.com/Automattic/jetpack/blob/trunk/projects/js-packages/components/components/notice/style.module.scss) in JS Packages → Components. I should have double-checked against the Figma file once I had things in place though.

jsnmoon reviewed (approved):
Looks great; feel free to ship!

cc @jeffgolenski: The Odyssey Stats screenshot dimensions are 437px x 210px. Should we use a higher-resolution image here, or are we okay with a bit of pixelation?

Commit:
Merge branch 'trunk' of github.com:Automattic/jetpack into update/stats-28729-add-image-to-notice

a8ck3n commented:
> Looks great; feel free to ship!
> 
> cc @jeffgolenski: The Odyssey Stats screenshot dimensions are 437px x 210px. Should we use a higher-resolution image here, or are we okay with a bit of pixelation?

@jsnmoon We have higher resolution versions in the original issue.

I downsized them when I thought the max display height would be 210px. We could definitely resize from the original assets again if needed but we probably don't want to use the original assets as is. The gradient image itself was over 3 MB in size. I'll spin up a follow-up issue to review the images though, so we can see if there's a better size/quality trade-off we can hit.

a8ck3n committed:
Re-add original resolution promo image.

The original image is much sharper so adding that back for now. The gradient has been scaled down to introduce some space savings though.

a8ck3n commented:
**Update:** I restored the original promo image to address the fuzzy look Jason noted. We're currently still using a scaled version of the gradient though, as I'm not seeing a noticeable difference when restoring the original full size image file.

@jeffgolenski — Let me know if this works for you!

jsnmoon reviewed (approved):
Looks good!
